internal/controller/storageclaim_controller.go

```diff
@@ -424,6 +476,57 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 				if err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
 				}
+			case "VolumeReplicationClass":
+				data := map[string]string{}
+				err = json.Unmarshal(resource.Data, &data)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to unmarshal StorageClaim configuration response: %v", err)
+				}
+				data["replication.storage.openshift.io/replication-secret-namespace"] = r.OperatorNamespace
+				// generate a new clusterID for cephfs subvolumegroup, as
+				// storageclaim is clusterscoped resources using its
+				// hash as the clusterID
+				data["clusterID"] = r.storageClaimHash
+
+				// sanity check that there's a single DRClusterConfig per cluster
+				drclusterconfigs := &ramenv1alpha1.DRClusterConfigList{}
+				err = r.list(drclusterconfigs)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to get DRClusterConfigList: %v", err)
+				}
+				if len(drclusterconfigs.Items) > 1 {
+					return reconcile.Result{}, fmt.Errorf("there's more than one DRClusterConfig attached to the cluster")
+				}
+				// check if DRClusterConfig has been removed
+				if len(drclusterconfigs.Items) == 0 {
+					vrcs := &replicationv1alpha1.VolumeReplicationClassList{}
```

Comment:
just set the owneRef of VolumeReplicationClass as the drclusterconfig as both are clusterscoped resouces?